app/controllers/application_controller.rb

```diff
@@ -70,7 +70,13 @@ def store_referrer
   end
 
   def require_functional!
-    redirect_to edit_user_registration_path unless current_user.functional?
+    if current_user.functional?
+      nil
+    elsif !current_user.confirmed?
```

Comment:
Minor style thing -- could pull out this 5 line block to a `non_functional_user_setup_path` (or something) private method, and keep this filter as a one-liner. The explicit `nil` return above this feels odd.

Reply:
I agree, especially to the last bit. Maybe return early if the user is "functional"?